### PR TITLE
EVG-13246 Task in Spruce host page links to the wrong URL

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: src/pages/host/getHostEventString.tsx [hunk 1]
```diff
@@ -212,7 +212,7 @@ export const getHostEventString = (
           Assigned to run task{" "}
           <StyledLink
             data-cy="host-running-task-set-link"
-            href={`${getUiUrl()}/task${data.taskId}/${data.execution}`}
+            href={`${getUiUrl()}/task/${data.taskId}/${data.execution}`}
```
Comment (SupaJoon) [inline src/pages/host/getHostEventString.tsx:215]: are these links meant to point to spruce or the legacy ui?

Comment (malikchaya2) [reply src/pages/host/getHostEventString.tsx:215]: They currently link to legacy, but there is probably no reason for that. Should I switch them to Spruce?

Comment (malikchaya2) [reply src/pages/host/getHostEventString.tsx]: fixed

File: src/pages/host/Metadata.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const Metadata: React.FC<{
   const runningTaskId = runningTask?.id;
   const runningTaskName = runningTask?.name;
 
-  const taskLink = `${getUiUrl()}/task/${runningTaskId}`;
+  const taskLink = `/task/${runningTaskId}`;
```
Comment (khelif96) [inline src/pages/host/Metadata.tsx:29]: We have a utility function for this
https://github.com/evergreen-ci/spruce/blob/82684a550fe376bf2661be5fb74c4cd911941707/src/constants/routes.ts#L94

File: src/pages/host/getHostEventString.tsx [hunk 3]
```diff
@@ -225,7 +224,7 @@ export const getHostEventString = (
           Current running task cleared (was:
           <StyledLink
             data-cy="host-running-task-cleared-link"
-            href={`${getUiUrl()}/task${data.taskId}/${data.execution}`}
+            href={`/task/${data.taskId}/${data.execution}`}
```
Comment (khelif96) [inline src/pages/host/getHostEventString.tsx:227]: Could you use the utility function i mentioned above for these use cases. You will need to modify it to also take in execution as an optional paramater

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: src/pages/host/getHostEventString.tsx [hunk 4]
```diff
@@ -212,7 +211,7 @@ export const getHostEventString = (
           Assigned to run task{" "}
           <StyledLink
             data-cy="host-running-task-set-link"
-            href={`${getUiUrl()}/task${data.taskId}/${data.execution}`}
+            href={`/task/${data.taskId}/${data.execution}`}
```
Comment (khelif96) [inline src/pages/host/getHostEventString.tsx:214]: Also this url pattern isn't correct for tasks.
Task urls follow the following pattern
`/task/:id/:tab?url_params`
url_params being `execution` in this case

File: src/pages/host/getHostEventString.tsx (hunk 3 above)
Comment (malikchaya2) [reply src/pages/host/getHostEventString.tsx:227]: I actually won't use execution because it's null

**Review by khelif96 — APPROVED** [review]

**Pull request merged**